src/Microsoft.ML.Data/Transforms/KeyToValueTransform.cs

```diff
@@ -67,386 +79,553 @@ private static VersionInfo GetVersionInfo()
-                    types[iinfo] = typeVals.ItemType;
-                else
-                    types[iinfo] = new VectorType(typeVals.ItemType.AsPrimitive, typeSrc.AsVector);
-                Func<int, ColumnType, ColumnType, KeyToValueMap> func = GetKeyMetadata<int, int>;
-                var meth = func.GetMethodInfo().GetGenericMethodDefinition().MakeGenericMethod(new Type[] { typeSrc.ItemType.RawType,
-                    types[iinfo].ItemType.RawType });
-                kvMaps[iinfo] = (KeyToValueMap)meth.Invoke(this, new object[] { iinfo, typeSrc, typeVals });
+                _parent = parent;
+                ComputeKvMaps(inputSchema, out _types, out _kvMaps);
             }
-            md.Seal();
-        }
 
-        private KeyToValueMap GetKeyMetadata<TKey, TValue>(int iinfo, ColumnType typeKey, ColumnType typeVal)
-        {
-            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
-            Host.AssertValue(typeKey);
-            Host.AssertValue(typeVal);
-            Host.Assert(typeKey.ItemType.RawType == typeof(TKey));
-            Host.Assert(typeVal.ItemType.RawType == typeof(TValue));
-
-            var keyMetadata = default(VBuffer<TValue>);
-            Source.Schema.GetMetadata<VBuffer<TValue>>(MetadataUtils.Kinds.KeyValues, Infos[iinfo].Source, ref keyMetadata);
-            Host.Check(keyMetadata.Length == typeKey.ItemType.KeyCount);
-
-            VBufferUtils.Densify<TValue>(ref keyMetadata);
-            return new KeyToValueMap<TKey, TValue>(this, typeKey.ItemType.AsKey, typeVal.ItemType.AsPrimitive, keyMetadata.Values, iinfo);
-        }
+            public bool CanSavePfa => true;
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
-        {
-            Host.Assert(0 <= iinfo & iinfo < _types.Length);
-            var type = _types[iinfo];
-            Host.Assert(type != null);
-            return type;
-        }
-
-        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
-        {
-            Host.AssertValueOrNull(ch);
-            Host.AssertValue(input);
-            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
-            disposer = null;
+            public override RowMapperColumnInfo[] GetOutputColumns()
+            {
+                var result = new RowMapperColumnInfo[_parent.ColumnPairs.Length];
+                for (int i = 0; i < _parent.ColumnPairs.Length; i++)
+                {
+                    var meta = RowColumnUtils.GetMetadataAsRow(InputSchema, ColMapNewToOld[i],
+                        x => x == MetadataUtils.Kinds.SlotNames);
+                    result[i] = new RowMapperColumnInfo(_parent.ColumnPairs[i].output, _types[i], meta);
+                }
+                return result;
+            }
 
-            return _kvMaps[iinfo].GetMappingGetter(input);
-        }
+            public void SaveAsPfa(BoundPfaContext ctx)
+            {
+                throw new NotImplementedException();
```

Comment:
> throw new NotImplementedException(); [](start = 15, length = 37)

 Host.CheckValue(ctx, nameof(ctx));

                var toHide = new List<string>();
                var toDeclare = new List<KeyValuePair<string, JToken>>();

                for (int iinfo = 0; iinfo < _infos.Length; ++iinfo)
                {
                    var info = _infos[iinfo];
                    var srcName = info.Source;
                    string srcToken = ctx.TokenOrNullForName(srcName);
                    if (srcToken == null)
                    {
                        toHide.Add(info.Name);
                        continue;
                    }
                    var result = _kvMaps[iinfo].SavePfa(ctx,srcToken);
                    if (result == null)
                    {
                        toHide.Add(info.Name);
                        continue;
                    }
                    toDeclare.Add(new KeyValuePair<string, JToken>(info.Name, result));
                }
                ctx.Hide(toHide.ToArray());
                ctx.DeclareVar(toDeclare.ToArray());